rest/server_context.go

```
@@ -377,18 +378,18 @@ func (sc *ServerContext) ReloadDatabaseWithConfig(nonContextStruct base.NonCance
 
 func (sc *ServerContext) _reloadDatabaseWithConfig(ctx context.Context, config DatabaseConfig, failFast bool) error {
 	sc._removeDatabase(ctx, config.Name)
-	_, err := sc._getOrAddDatabaseFromConfig(ctx, config, false, db.GetConnectToBucketFn(failFast))
+	_, err := sc._getOrAddDatabaseFromConfig(ctx, config, false, db.GetConnectToBucketFn(failFast), failFast)
 	return err
 }
 
 // Adds a database to the ServerContext.  Attempts a read after it gets the write
 // lock to see if it's already been added by another process. If so, returns either the
 // existing DatabaseContext or an error based on the useExisting flag.
-func (sc *ServerContext) getOrAddDatabaseFromConfig(ctx context.Context, config DatabaseConfig, useExisting bool, openBucketFn db.OpenBucketFn) (*db.DatabaseContext, error) {
+func (sc *ServerContext) getOrAddDatabaseFromConfig(ctx context.Context, config DatabaseConfig, useExisting bool, openBucketFn db.OpenBucketFn, failFast bool) (*db.DatabaseContext, error) {
```

I think we're using openBucketFn exclusively to determine whether to fail fast on that operation, so adding a separate failFast flag seems potentially redundant. I think the motivation is the same for both cases - whether we should retry on NotFound when (a) the bucket doesn't exist, and (b) a collection doesn't exist.

We only ever invoke this with GetConnectToBucketFn for openBucketFn for non-test usages, so for those cases it would be preferable to just pass the failFast flag instead of an openBucketFn AND failFast flag.

I understand not wanting to bloat this PR, but it's a bit ugly that every non-test invocation of this function looks like `db.GetConnectToBucketFn(failFast), failFast`